pulsar-broker/src/main/java/org/apache/pulsar/broker/delayed/InMemoryDelayedDeliveryTracker.java

```diff
@@ -126,10 +126,13 @@ public boolean addMessage(long ledgerId, long entryId, long deliverAt) {
         }
 
         long timestamp = trimLowerBit(deliverAt, timestampPrecisionBitCnt);
-        delayedMessageMap.computeIfAbsent(timestamp, k -> new Long2ObjectRBTreeMap<>())
-                .computeIfAbsent(ledgerId, k -> new Roaring64Bitmap())
-                .add(entryId);
-        delayedMessagesCount.incrementAndGet();
+        Roaring64Bitmap roaring64Bitmap = delayedMessageMap
+                .computeIfAbsent(timestamp, k -> new Long2ObjectRBTreeMap<>())
+                .computeIfAbsent(ledgerId, k -> new Roaring64Bitmap());
+        if (!roaring64Bitmap.contains(entryId)) {
+            roaring64Bitmap.add(entryId);
```

Comment:
It looks like `.addLong` should be used in the Roaring64Bitmap API.
```suggestion
            roaring64Bitmap.addLong(entryId);
```
The `.add` method works too, but the method signature takes a long array (long...). Perhaps the compiler is able to optimize that, so it might not make a difference.

It's unfortunate that `Roaring64Bitmap` doesn't have the `checkedAdd` method as there is in `RoaringBitmap`. That would eliminate the need for the `.contains` check.